File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public static int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> occ = new HashMap<>();
        int n = nums.length;
        for(int i = 0; i<n ; i++){
            if(occ.containsKey(nums[i]))
                occ.put(nums[i], occ.get(nums[i]) +1);
            else occ.put(nums[i], 1);
        }
        Queue<Map.Entry<Integer,Integer>> queue = new PriorityQueue<>((a,b) -> b.getValue() - a.getValue());
        for(Map.Entry<Integer,Integer> element:occ.entrySet()){
            queue.add(element);
        }
        int[] res = new int[k];
        for(int i = 0 ; i<k; i++){
            res[i] = queue.poll().getKey();
        }
        return res;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.ts
function topKFrequent(nums: number[], k: number): number[] {
    const map = new Map<number,number>();
    for(const num of nums){
        map.set(num, (map.get(num) || 0) + 1);
    }
    return Array.from(map.entries())
                .sort((val1, val2) => val2[1]-val1[1])
                .slice(0,k)
                .map(entry => entry[0]);
};